Chelone.py

```diff
@@ -0,0 +1,240 @@
+alternative to pyGame with more of a focus on sprites rather then primitives
+
+The engine includes basic collision detection, animation, drawing, z-layering and complex backrounds
+'''
+import tkinter as tk
+from typing import Union, Callable
+from time import time, sleep
+
+TARGET_FPS = 60
+InFocus = True
+
+def init(resolution_x:int, resolution_y:int):
+	# Create the window with the Tk class
+	root = tk.Tk()
+	
+	root.bind("<Key>", on_key_press)
+	root.bind("<KeyRelease>", on_key_release)
+	root.bind("<FocusIn>", FocusIn)
+	root.bind("<FocusOut>", FocusOut)
+
+	# Create the canvas and make it visible with pack()
+	canvas = tk.Canvas(root, width=resolution_x, height=resolution_y)
+	canvas.pack() # this makes it visible
+
+	clear_img = tk.PhotoImage(file="sprites/clear.png")
+
+	return SpriteRenderer(root,canvas)
+
+
+def FocusIn(event):
+	global InFocus
+	InFocus = True
+
+# Function that handles cleaning up our input after window goes out of focus
+def FocusOut(event):
+	global InFocus, pressed_keys, caught_keys
+	InFocus = False
+	pressed_keys = []
+	caught_keys = []
+
+
+caught_keys_prev_prev = []
+caught_keys_prev = []
+caught_keys= []
+
+pressed_keys = []
+
+clear_img = None
+
+def check_keys():
+	global caught_keys_prev, caught_keys_prev_prev, caught_keys, pressed_keys
+	# remove keys not caught for 3 frames
+	for key in pressed_keys:
+		if key not in caught_keys and\
+		  key not in caught_keys_prev and\
+		  key not in caught_keys_prev_prev:
+
+			pressed_keys.remove(key)
+
+	for key in caught_keys:
+		if key not in pressed_keys:
+			pressed_keys.append(key)
+
+	caught_keys_prev_prev = caught_keys.copy()
+
+def on_key_press(event):
+	global caught_keys
+	caught_keys.append(event.keysym)
+
+
+def on_key_release(event):
+	global caught_keys
+	caught_keys.remove(event.keysym)
+
+
+
+class Sprite():
+	"""Object that contains data about the sprite. Also can be used to have code ran on each frame"""
+	x = 0
+	y = 0
+	var = {}
+	image = None
+	image_tk = None
+	parent_canvas = None
+	updateFunc = None
+	_passedUpdateFunc=None
+	
+	def __init__(self, image:Union[str, tk.PhotoImage], x:int=0, y:int=0, updateFunc:Callable[..., None]=lambda a:None, setupFunc:Callable[..., None]=lambda a:None):
+		self.x = x
+		self.y = y
+		self.var = {}
+
+		if isinstance(image, str):
+			self.image = tk.PhotoImage(file=image)
+		else:
+			self.image = image
+		self.image_tk = None
+		self.parent_canvas = None
+		self._passedUpdateFunc = updateFunc
+		self.updateFunc = updateFunc
+
+		setupFunc(self)
+
+	def move(self, x:int, y:int):
+		self.x += x
+		self.y += y
+		self.parent_canvas.move(self.image_tk, x, y)
+
+	def change_image(self, image:tk.PhotoImage):
+		self.image = image
+		self.parent_canvas.itemconfig(self.image_tk, image=self.image)
+
+	def shedule_anim(self, anim_frames:list):
+		global clear_img
+
+		i = 0
+		tmp = self._passedUpdateFunc
+		
+		def anim_sheduler(sprite):
+			nonlocal i, tmp
+
+			sprite.change_image(anim_frames[i])
+			i+=1
```

Comment:
binary operators must be indentated.

Reply:
Please elaborate. What is indentated?

Reply:
My bad, indented.
Surrounded, edged, or divided  by some symbol or a whitespace in this case